app/src/main/java/ch/deletescape/lawnchair/QsbBlockerView.java

```diff
@@ -136,18 +130,25 @@ public void onClick(View view) {
                             getContext().startActivity(intent);
                         }
                     });
-                    OpenWeatherMapHelper.CurrentWeatherCallback callback = new OpenWeatherMapHelper.CurrentWeatherCallback() {
-                        @Override
-                        public void onSuccess(CurrentWeather currentWeather) {
-                            temperature.setText(currentWeather.getMain().getTemp() + (isImperial ? "°F" : "°C"));
-                        }
-
-                        @Override
-                        public void onFailure(Throwable throwable) {
-                            temperature.setText("ERROR°C");
-                        }
-                    };
-                    helper.getCurrentWeatherByCityName(city, callback);
+                    SharedPreferences prefs = Utilities.getPrefs(getContext());
+                    String apiKey = prefs.getString("pref_weatherDebug_api_key", BuildConfig.OPENWEATHERMAP_KEY);
+                    String city = prefs.getString("pref_weatherDebug_city", "Lucerne, CH");
+                    String units = prefs.getString("pref_weatherDebug_units", "metric");
+                    final boolean isImperial = units.equals("imperial");
+                    new OWMWeatherDownload()
+                            .setAPIKey(apiKey)
+                            .setLocation(city)
+                            .setWeatherUnits(isImperial ? WeatherUnits.IMPERIAL : WeatherUnits.METRIC)
```

Comment:
here u can use setWeatherUnits(units) instead of check if units is equeal "imperial" and then set again check if isImperial then use WeatherUnits.IMPERIAL ...